torch/nn/modules/rnn.py

```diff
@@ -123,11 +125,16 @@ def __init__(self, mode: str, input_size: int, hidden_size: int,
                 self._flat_weights_names.extend(param_names)
                 self._all_weights.append(param_names)
 
-        self._flat_weights = [(lambda wn: getattr(self, wn) if hasattr(self, wn) else None)(wn) for wn in self._flat_weights_names]
-        self.flatten_parameters()
+        self._init_flat_weights()
 
         self.reset_parameters()
 
+    def _init_flat_weights(self):
+        self._flat_weights = [(lambda wn: getattr(self, wn) if hasattr(self, wn) else None)(wn) for wn in self._flat_weights_names]
```

Comment:
Not yours but:
```
self._flat_weights = [getattr(self, wn, None) for wn in self._flat_weights_names]
```

Reply:
Haha good call, super weird to use a lambda this way

Reply:
So mypy does not like the use of `getattr(self, wn, None)`. i can still get rid of the lambda though